test/miner_ct_utils.erl

```diff
@@ -1230,6 +1435,10 @@ new_random_key(Curve) ->
     #{secret := PrivKey, public := PubKey} = libp2p_crypto:generate_keys(Curve),
     {PrivKey, PubKey}.
 
+new_random_key_with_sig_fun(Curve) ->
+    #{secret := PrivKey, public := PubKey} = libp2p_crypto:generate_keys(Curve),
+    SigFun = libp2p_crypto:mk_sig_fun(PrivKey),
+    {PrivKey, PubKey, SigFun}.
```

Comment:
`t_user` module is an abstraction to address this use case.

Reply:
@xandkar t_user mod exists only in blockchain core.  Not available for miner tests